src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/ChangeTrackingServiceTest.kt

```diff
@@ -0,0 +1,296 @@
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.locations.LocationsService
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.mapping.toModel
+import java.time.LocalTime
+import java.util.Optional
+import java.util.UUID
+
+class ChangeTrackingServiceTest {
+  private val videoBookingRepository: VideoBookingRepository = mock()
+  private val locationsService: LocationsService = mock()
+  private val additionalBookingDetailRepository: AdditionalBookingDetailRepository = mock()
+  private val service = ChangeTrackingService(videoBookingRepository, locationsService, additionalBookingDetailRepository)
+
+  @Nested
+  inner class HasCourtBookingChanged {
+    private val existingCourtBooking = courtBooking(
+      hearingType = CourtHearingType.TRIBUNAL,
+      cvpLinkDetails = CvpLinkDetails.url("https://video.link.com"),
+      notesForStaff = "staff notes",
+      notesForPrisoners = "prisoner notes",
+    ).withMainCourtPrisonAppointment(
+      prisonCode = WANDSWORTH,
+      location = wandsworthLocation,
+      date = today(),
+      startTime = LocalTime.of(10, 0),
+      endTime = LocalTime.of(11, 0),
+    )
+
+    private val amendCourtBookingRequest = amendCourtBookingRequest(
+      prisonCode = WANDSWORTH,
+      location = wandsworthLocation,
+      appointmentDate = today(),
+      startTime = LocalTime.of(10, 0),
+      endTime = LocalTime.of(11, 0),
+      hearingType = CourtHearingType.TRIBUNAL,
+      cvpLinkDetails = CvpLinkDetails.url("https://video.link.com"),
+      notesForStaff = "staff notes",
+    )
+
+    @Test
+    fun `should be no changes as court user`() {
+      whenever(videoBookingRepository.findById(1L)) doReturn Optional.of(existingCourtBooking)
+      whenever(locationsService.getLocationByKey(amendCourtBookingRequest.prisoners.single().appointments.single().locationKey!!)) doReturn wandsworthLocation.toModel()
+
+      service.hasBookingChanged(1L, amendCourtBookingRequest, COURT_USER) isEqualTo false
+    }
+
+    @Test
+    fun `should fail when booking not found as court user`() {
+      whenever(videoBookingRepository.findById(1L)) doReturn Optional.empty()
+
+      val error = assertThrows<EntityNotFoundException> { service.hasBookingChanged(1L, amendCourtBookingRequest, COURT_USER) }
+      error.message isEqualTo "Video booking with ID 1 not found."
+    }
+
+    @Test
+    fun `should fail when booking types do not match as court user`() {
+      whenever(videoBookingRepository.findById(1L)) doReturn Optional.of(probationBooking())
+
+      val error = assertThrows<IllegalArgumentException> { service.hasBookingChanged(1L, amendCourtBookingRequest, COURT_USER) }
+      error.message isEqualTo "Request type and existing booking type must be the same. Request type is COURT and booking type is PROBATION."
+    }
+
+    @Test
+    fun `should be no changes for prisoner notes as court user`() {
```

Comment:
Technically prisoner notes would not be supplied by a court user (same for probation).  However this ensures it.